### PR TITLE
GO-2512 Update backlinks on links creation

### PR DESCRIPTION
**Pull request by KirillSto**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

This PR fixes backlinks update for objects that are already opened

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2512/backlinks-are-not-updated-in-all-usecases

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

File: core/block/editor/smartblock/links.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+				return b.Apply(b.NewState().SetDetails(details), KeepInternalFlags)
+			}); err != nil {
+				log.With("objectID", sb.Id()).Errorf("failed to update backlinks for object %s: %v", id, err)
+			}
+		}
+	}
+}
+
+func (sb *smartBlock) navigationalLinks(s *state.State) []string {
+	includeRelations := sb.includeRelationObjectsAsDependents
```
Comment (fat-fellow) [inline core/block/editor/smartblock/links.go:98]: separate func, too long

Comment (KirillSto) [reply core/block/editor/smartblock/links.go]: Split into two separate functions

File: pkg/lib/localstore/objectstore/update.go [hunk 2]
```diff
@@ -106,6 +98,58 @@ func (s *dsObjectStore) UpdatePendingLocalDetails(id string, proc func(details *
...
+		oldDetails := objDetails.GetDetails()
+		if oldDetails == nil || oldDetails.Fields == nil {
+			oldDetails = &types.Struct{Fields: map[string]*types.Value{}}
+		}
+		newDetails, err := proc(oldDetails)
+		if err != nil {
+			return fmt.Errorf("run a modifier: %w", err)
+		}
+
+		if newDetails == nil {
```
Comment (AnastasiaShemyakinskaya) [inline pkg/lib/localstore/objectstore/update.go:123]: Could be simplified like this 
```
if newDetails == nil || newDetails.Fields == nil {
	newDetails = &types.Struct{
		Fields: map[string]*types.Value{},
	}
}
```

Comment (KirillSto) [reply pkg/lib/localstore/objectstore/update.go]: Thank you! Fixed

**Review by requilence — APPROVED** [review]

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Pull request merged**